commit-event-service/src/main/scala/ch/datascience/commiteventservice/events/categories/commitsync/eventgeneration/CommitEventSynchronizer.scala

```diff
@@ -0,0 +1,258 @@
+    case (elapsedTime, Existed) =>
+      logger.info(logMessageFor(eventId, project, s"no new events found in ${elapsedTime}ms"))
+    case (elapsedTime, Created) =>
+      logger.info(logMessageFor(eventId, project, s"new events found in ${elapsedTime}ms"))
+    case (elapsedTime, Deleted) =>
+      logger.info(
+        logMessageFor(eventId, project, s"events found for deletion in ${elapsedTime}ms")
+      )
+    case (elapsedTime, Failed(message, exception)) =>
+      logger.error(exception)(logMessageFor(eventId, project, s"$message in ${elapsedTime}ms"))
+  }
+
+  private def logSummary(eventId: CommitId,
+                         project: Project
+  ): ((ElapsedTime, SynchronizationSummary)) => Interpretation[Unit] = { case (elapsedTime, summary) =>
+    logger.info(
+      logMessageFor(
+        eventId,
+        project,
+        s"events generation result: ${summary.created} created, ${summary.existed} existed, ${summary.skipped} skipped, ${summary.deleted} deleted, ${summary.failed} failed in ${elapsedTime}ms"
+      )
+    )
+
+  }
+
+  private def logMessageFor(eventId: CommitId, project: Project, message: String) =
+    s"$categoryName: id = $eventId, projectId = ${project.id}, projectPath = ${project.path} -> $message"
+
+}
+
+private[commitsync] object CommitEventSynchronizer {
+  def apply(gitLabThrottler:       Throttler[IO, GitLab],
+            executionTimeRecorder: ExecutionTimeRecorder[IO],
+            logger:                Logger[IO]
+  )(implicit
+      executionContext: ExecutionContext,
+      contextShift:     ContextShift[IO],
+      timer:            Timer[IO]
+  ) = for {
+    accessTokenFinder  <- AccessTokenFinder(logger)
+    latestCommitFinder <- LatestCommitFinder(gitLabThrottler, logger)
+    eventDetailsFinder <- EventDetailsFinder(logger)
+    commitInfoFinder   <- CommitInfoFinder(gitLabThrottler, logger)
+    commitToEventLog   <- CommitToEventLog(logger)
+    commitEventRemover <- CommitEventsRemover(logger)
+
+  } yield new CommitEventSynchronizerImpl[IO](accessTokenFinder,
+                                              latestCommitFinder,
+                                              eventDetailsFinder,
+                                              commitInfoFinder,
+                                              commitToEventLog,
+                                              commitEventRemover,
+                                              executionTimeRecorder,
+                                              logger
+  )
+
+  sealed trait UpdateResult extends Product with Serializable
+  object UpdateResult {
+    final case object Skipped extends UpdateResult
+    final case object Created extends UpdateResult
+    final case object Existed extends UpdateResult
+    final case object Deleted extends UpdateResult
+    case class Failed(message: String, exception: Throwable) extends UpdateResult
+  }
+
+  final case class SynchronizationSummary(skipped: Int = 0,
```

Comment:
I'm just wondering if a simple Map[Update, Int] could be enough for keeping the counts. However, I feel I might lacking some info to be sure if that's a better solution.